### PR TITLE
Beans definition and overriding poc (DO NOT MERGE)

### PR DESCRIPTION
**Pull request by ArturGajowy**
See tests first, then ServiceRestClientConfiguration for implementation requirements and approach.

### CODE REVIEW

File: micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientConfiguration.groovy [hunk 1]
```diff
@@ -45,11 +42,31 @@ class ServiceRestClientConfiguration {
...
+     *
+     * The above can be achieved easily by creating an ast transformation that will add
+     * `@Qualifier('micro-infra')` to all method parameters (and - optionally - all @Bean definitions).
+     * The transformation can also report compile errors when the bean name does not start with 'microInfra'.
+     *
+     * Adding @ConditionalOnMissingBean can be done either by the transformation or by using a meta-annotation
+     * that will carry all of [@Bean, @Qualifier('micro-infra-spring'), @ConditionalOnMissingBean] declarations.
+     */
     @Bean
+    @Qualifier('micro-infra-spring')
```
Comment (mariuszs) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientConfiguration.groovy]: Dedicated annotation for type safety

    @Qualifier('micro-infra-spring')
    @Retention(RUNTIME)
    @Target({TYPE, METHOD, FIELD, PARAMETER})
    public @interface MicroInfraSpring {}

Comment (ArturGajowy) [inline micro-infra-spring-base/src/main/groovy/com/ofg/infrastructure/web/resttemplate/fluent/ServiceRestClientConfiguration.groovy]: Sure thing! :)

**Comment by marcingrzejszczak** [comment]
IMO this solution (https://github.com/4finance/micro-infra-spring/pull/179) is much cleaner. WDYT?

**Comment by marcingrzejszczak** [comment]
Can we close this, or what?

**Comment by marcingrzejszczak** [comment]
Can we close this?

**Pull request closed without merging**